src/templates/finops-hub/modules/dataFactory.bicep

```diff
@@ -0,0 +1,715 @@
+// Queuing up the transform pipeline and exiting immediately greatly reduces the likelihood of this happening.
+//------------------------------------------------------------------------------
+
+// Get storage account instance
+resource storageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' existing = {
+  name: storageAccountName
+}
+
+// Create trigger
+resource trigger_exportContainer 'Microsoft.DataFactory/factories/triggers@2018-06-01' = {
+  name: safeExportContainerName
+  parent: dataFactory
+  dependsOn: [
+    stopHubTriggers
+    pipeline_extractExport
+  ]
+  properties: {
+    annotations: []
+    pipelines: [
+      {
+        pipelineReference: {
+          referenceName: '${exportContainerName}_extract'
+          type: 'PipelineReference'
+        }
+        parameters: {
+          folderName: '@triggerBody().folderPath'
+          fileName: '@triggerBody().fileName'
+        }
+      }
+    ]
+    type: 'BlobEventsTrigger'
+    typeProperties: {
+      blobPathBeginsWith: '/${exportContainerName}/blobs/'
+      blobPathEndsWith: '.csv'
+      ignoreEmptyBlobs: true
+      scope: storageAccount.id
+      events: [
+        'Microsoft.Storage.BlobCreated'
+      ]
+    }
+  }
+}
+
+resource pipeline_extractExport 'Microsoft.DataFactory/factories/pipelines@2018-06-01' = {
+  name: '${safeExportContainerName}_extract'
+  parent: dataFactory
+  dependsOn: [
+    pipeline_transformExport
+  ]
+  properties: {
+    activities: [
+      {
+        name: 'Execute'
+        type: 'ExecutePipeline'
+        dependsOn: []
+        userProperties: []
+        typeProperties: {
+          pipeline: {
+            referenceName: '${safeExportContainerName}_transform'
+            type: 'PipelineReference'
+          }
+          waitOnCompletion: false
+          parameters: {
+            folderName: {
+              value: '@pipeline().parameters.folderName'
+              type: 'Expression'
+            }
+            fileName: {
+              value: '@pipeline().parameters.fileName'
+              type: 'Expression'
+            }
+          }
+        }
+      }
+    ]
+    parameters: {
+      folderName: {
+        type: 'string'
+      }
+      fileName: {
+        type: 'string'
+      }
+    }
+    annotations: []
+  }
+}
+
+//------------------------------------------------------------------------------
+// Export container transform pipeline
+// Trigger: pipeline_extractExport
+//
+// Converts CSV files to Parquet or .CSV.GZ files.
+//------------------------------------------------------------------------------
+
+resource pipeline_transformExport 'Microsoft.DataFactory/factories/pipelines@2018-06-01' = {
+  name: '${safeExportContainerName}_transform'
+  parent: dataFactory
+  dependsOn: [
+    dataset_msexports
+    dataset_ingestion
+  ]
+  properties: {
+    activities: [
+      // (start) -> Wait -> Scope -> Metric -> Date -> File -> Folder -> Delete Target -> Convert CSV -> Delete CSV -> (end)
```

Comment:
FYI - I added this to help document the code. It's not sent to ADF. The other comments above each step are just to identify which section is which when collapsed.